CSharp/Library/Microsoft.Bot.Connector.Shared/ActivityEx.cs

```diff
@@ -149,17 +147,57 @@ public Activity CreateReply(string text = null, string locale = null)
         public IInvokeActivity AsInvokeActivity() { return IsActivity(ActivityTypes.Invoke) ? this : null; }
 
         /// <summary>
+        public static string GetActivityType(string type)
+        {
+            if (String.Equals(type, ActivityTypes.Message, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.Message;
+
+            if (String.Equals(type, ActivityTypes.ContactRelationUpdate, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.ContactRelationUpdate;
+
+            if (String.Equals(type, ActivityTypes.ConversationUpdate, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.ConversationUpdate;
+
+            if (String.Equals(type, ActivityTypes.DeleteUserData, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.DeleteUserData;
+
+            if (String.Equals(type, ActivityTypes.Typing, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.Typing;
+
+            if (String.Equals(type, ActivityTypes.Ping, StringComparison.OrdinalIgnoreCase))
+                return ActivityTypes.Ping;
+
+            return $"{Char.ToLower(type[0])}{type.Substring(1)}";
+        }
+
+        public bool HasContent()
+        {
+            // hit the extension method

```

Comment:
`IMessageActivity` requires `HasContent()` and `GetMentions()` - didn't want to go changing an interface.